### ISSUE TITLE
[Bug]: massCode UI and search gets slower if you have snippets with long lists of numbers

### ISSUE THREAD

**Issue by joyofscripting** [massCodeIO/massCode#270]
### Describe the bug

I discovered that my beloved installation of massCode gets a bit unresponsive and takes a longer time for searches when I add SQL snippets that contain long lists of comma-separated numbers in the WHERE-condition.

Normally massCode is very responsive. you can click anywhere and the UI will react immediately. Also searches are normally very fast, a real joy. But when you add a snippet of the mentioned type, massCode's UI will become less responsive, searches will take some time and while the search is performed, the UI is almost blocked.

Currently I have 216 snippets organized in massCode, mostly short SQL statements, Python and Swift code.

Of couse one could argue that a snippet with a list of 1000 comma-separated values is not really a snippet anymore. And I can absolutely understand this perspective. But it was working perfectly in previous versions. So it might be just somd kind of bug.

### To reproduce

1. Add a new snippet with the title 'SQLTEST' in masscode
2. Genererate a long list (>= 1000) of comma-separated numbers (see example file below) and paste them into the newly created snippet
3. Set the syntax format of the snippet to SQL 
4. Select another snippet of your library
5. Search for 'SQLTEST'

In my case the UI gets unresponsive after step 5 and the search takes some time (10 seconds).

[list_of_1000_numbers.txt](https://github.com/massCodeIO/massCode/files/9453990/list_of_1000_numbers.txt)

### Version

3.4.1

### OS

Windows 10

### Validations

- [X] Follow our [Code of Conduct](https://github.com/massCodeIO/massCode/blob/master/CODE_OF_CONDUCT.md)
- [X] Check that there isn't [already an issue](https://github.com/massCodeIO/massCode/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/massCodeIO/massCode/discussions).

**Comment by antonreshetov** [comment]
@joyofscripting The previous versions in which everything was fine, is it below 3? Since the breaking change occurred in version 3, in which there was a switch from ace to codemiror and texmate grammars

**Comment by joyofscripting** [comment]
Yes, this might be true.  I remember that it worked back in July and as far as I know version 3 saw the world at the beginning of August,

**Comment by antonreshetov** [comment]
I originally thought it might be related to [onigasm](https://github.com/zikaari/onigasm) and it seems to be right. I found this issue https://github.com/zikaari/onigasm/issues/6. Unfortunately the issue is not resolved. 

I also tried replacing the `sql` grammar, the result is much better, but there are problems with the syntax highlighting in `sqlserver`

Original grammar -  https://raw.githubusercontent.com/microsoft/vscode/main/extensions/sql/syntaxes/sql.tmLanguage.json

New grammar-  https://github.com/textmate/sql.tmbundle/blob/master/Syntaxes/SQL.plist

**Comment by RedCMD** [comment]
here's my VSCode extension to help with authoring TextMate grammars https://marketplace.visualstudio.com/items?itemName=RedCMD.tmlanguage-syntax-highlighter

it can help with debugging performance issues

it's prob a issue with a regex matching a lot of text in the line and then failing at the end.
moving one char forward and trying again.
leading to enormous amount of memory reads.
prob just needs a quick-fail at the begin of the regex eg `\b` or `(?<=\s)` etc

**Comment by antonreshetov** [comment]
**Closing as outdated.**

This issue was reported against an older version of massCode (v3/v4) that relied on the legacy storage system (`db.json` / SQLite). massCode v5 introduces a completely new architecture based on a markdown vault, and we no longer maintain or fix issues for versions below v5.

If you can still reproduce this on the latest v5 release, please open a new issue with up-to-date reproduction steps. Thanks for your report and for using massCode!
